### PR TITLE
executor: add more comments

### PR DESCRIPTION
**Pull request by zhexuany**

### CODE REVIEW

**Comment by shenli** [comment]
Please fix the CI. @zhexuany

File: executor/adapter.go [hunk 1]
```diff
@@ -155,17 +164,21 @@ type ExecStmt struct {
...
 
-// IsPrepared implements ast.Statement interface.
+// IsPrepared return true if stmt is already prepared.
 func (a *ExecStmt) IsPrepared() bool {
 	return a.isPreparedStmt
 }
 
-// IsReadOnly implements ast.Statement interface.
+// IsReadOnly return true if stmt is read only.
+// First fetch a to be checked statement from ExecStme
```
Comment (shenli) [inline executor/adapter.go]: Can not understand this sentence.

File: executor/adapter.go [hunk 2]
```diff
@@ -174,7 +187,10 @@ func (a *ExecStmt) IsReadOnly() bool {
 	return ast.IsReadOnly(readOnlyCheckStmt)
 }
 
-// RebuildPlan implements ast.Statement interface.
+// RebuildPlan rebuilds current execut statement plan.
+// It first gets all related schme info from context, and then preprocess
```
Comment (shenli) [inline executor/adapter.go]: `schme` is a typo.

File: executor/adapter.go [hunk 3]
```diff
@@ -155,17 +164,21 @@ type ExecStmt struct {
...
-// IsPrepared implements ast.Statement interface.
+// IsPrepared return true if stmt is already prepared.
 func (a *ExecStmt) IsPrepared() bool {
 	return a.isPreparedStmt
 }
 
-// IsReadOnly implements ast.Statement interface.
+// IsReadOnly return true if stmt is read only.
+// First fetch a to be checked statement from ExecStme
+// and update it if plan is a execute plan.
```
Comment (shenli) [inline executor/adapter.go]: This detailed info should be commented in the function, not as the comment of the function.

File: executor/distsql.go [hunk 4]
```diff
@@ -346,6 +349,11 @@ func (e *TableReaderExecutor) Next(goCtx goctx.Context) (Row, error) {
 			e.feedback.Invalidate()
 			return nil, errors.Trace(err)
 		}
+		//TODO: isn't better we let its caller handle this situation?
```
Comment (zhexuany) [inline executor/distsql.go]: @shenli PTAL here.

File: executor/adapter.go (hunk 3 above)
Comment (zhexuany) [reply executor/adapter.go]: In my opinion, a function's comment should describe basic logic of the function. 
here is a example https://github.com/golang/go/blob/master/src/fmt/print.go#L411.

File: executor/distsql.go [hunk 5]
```diff
@@ -97,6 +97,7 @@ func (task *lookupTableTask) getRow(schema *expression.Schema) (Row, error) {
 	return nil, nil
 }
 
+//
```
Comment (winoros) [inline executor/distsql.go]: Fix this.

**Comment by shenli** [comment]
@zhexuany Please address the comments.

**Comment by zhexuany** [comment]
@shenli PTAL just resolve reviewer's comment.

**Comment by shenli** [comment]
LGTM

File: executor/adapter.go [hunk 6]
```diff
@@ -80,6 +80,9 @@ func schema2ResultFields(schema *expression.Schema, defaultDB string) (rfs []*as
 	return rfs
 }
 
+// Next uses recordSet's executor to get next available row.
+// If row is nil, then updates LastFoundRows in session variable.
+// If stmt and row are not nil, increase current FoundRows by 1.
 func (a *recordSet) Next(goCtx goctx.Context) (types.Row, error) {
```
Comment (tiancaiamao) [inline executor/adapter.go]: Next is depreciated. @zz-jason

File: executor/adapter.go [hunk 7]
```diff
@@ -155,17 +164,20 @@ type ExecStmt struct {
...
 }
 
-// OriginText implements ast.Statement interface.
+// OriginText returns original statement as a string.
 func (a *ExecStmt) OriginText() string {
 	return a.Text
 }
 
-// IsPrepared implements ast.Statement interface.
+// IsPrepared return true if stmt is already prepared.
```
Comment (tiancaiamao) [inline executor/adapter.go]: if stmt is a prepare statement.

File: executor/adapter.go [hunk 8]
```diff
@@ -99,6 +102,11 @@ func (a *recordSet) Next(goCtx goctx.Context) (types.Row, error) {
 	return row, nil
 }
 
+// NextChunk use uses recordSet's executor to get next available chunk for later usage.
+// If chunk do not contain any rows, then we update last query found rows in session variable as current found rows.
```
Comment (tiancaiamao) [inline executor/adapter.go]: s/do/does

File: executor/adapter.go [hunk 9]
```diff
@@ -174,8 +186,12 @@ func (a *ExecStmt) IsReadOnly() bool {
 	return ast.IsReadOnly(readOnlyCheckStmt)
 }
 
-// RebuildPlan implements ast.Statement interface.
+// RebuildPlan rebuilds current execut statement plan.
```
Comment (tiancaiamao) [inline executor/adapter.go]: s/execut/execute

File: executor/adapter.go [hunk 10]
```diff
@@ -174,8 +186,12 @@ func (a *ExecStmt) IsReadOnly() bool {
...
 func (a *ExecStmt) RebuildPlan() error {
+	// It first gets all related schmea info from context, and then preprocess
```
Comment (tiancaiamao) [inline executor/adapter.go]: I'm afraid this kind of comment is not intuitive than the code itself.

File: executor/distsql.go (hunk 4 above)
Comment (tiancaiamao) [reply executor/distsql.go]: Isn't it better if we ...

File: executor/distsql.go [hunk 11]
```diff
@@ -346,6 +348,11 @@ func (e *TableReaderExecutor) Next(goCtx goctx.Context) (Row, error) {
 			e.feedback.Invalidate()
 			return nil, errors.Trace(err)
 		}
+		//TODO: isn't better we let its caller handle this situation?
+		// If partial result return a nil row and a nil error, it means
+		// we already drained all result.
+		// To me, it is kind of wired using a loop to do the logic
```
Comment (tiancaiamao) [inline executor/distsql.go]: To subjective statement, note that TiDB is a cooperative project which is stared by many people, it's not a personal toy project.

Comment (tiancaiamao) [reply executor/distsql.go]: B.T.W, you can tiny refine the code to make it  more readable, as you like.

File: executor/distsql.go [hunk 12]
```diff
@@ -324,7 +322,11 @@ func (e *TableReaderExecutor) Close() error {
 	return errors.Trace(err)
 }
 
-// Next implements the Executor Next interface.
+// Next returns next available Row. In its process, any error will be returned.
+// It first try to fetch a partial result if current partial result is nil.
+// If any error appears during this stage, it simply return a error to its caller.
+// If it successfully initialize its partial result, it will use this to get next
+// available row.
 func (e *TableReaderExecutor) Next(goCtx goctx.Context) (Row, error) {
```
Comment (tiancaiamao) [inline executor/distsql.go]: `Next()` interface is depreciated.

File: server/conn.go [hunk 13]
```diff
@@ -974,10 +985,14 @@ func (cc *clientConn) writeColumnInfo(columns []*ColumnInfo) error {
 	return nil
 }
 
+// writeChunks writes data from a Chunk, which filled data by a ResultSet, into a a connection.
```
Comment (tiancaiamao) [inline server/conn.go]: into a a connection

File: server/conn.go [hunk 14]
```diff
@@ -974,10 +985,14 @@ func (cc *clientConn) writeColumnInfo(columns []*ColumnInfo) error {
...
+// binary specifies the way to dump data. It throws any error whiling dumping data.
```
Comment (tiancaiamao) [inline server/conn.go]: whiling

File: server/conn.go [hunk 15]
```diff
@@ -974,10 +985,14 @@ func (cc *clientConn) writeColumnInfo(columns []*ColumnInfo) error {
...
+// more will be passed into writeEOF.
 func (cc *clientConn) writeChunks(goCtx goctx.Context, rs ResultSet, binary bool, more bool) error {
 	data := make([]byte, 4, 1024)
 	chk := rs.NewChunk()
 	for {
+		// here server.tidbResultSet implements NextChunk method.
```
Comment (tiancaiamao) [inline server/conn.go]: The first word of a sentence should be capital when necessary.

File: executor/distsql.go (hunk 11 above)
Comment (zhexuany) [reply executor/distsql.go]: It is a question. I did not intend to leave this comment and merge into master. So per your understanding, is it normal here?

File: executor/adapter.go [hunk 16]
```diff
@@ -155,17 +164,20 @@ type ExecStmt struct {
...
 }
 
-// OriginText implements ast.Statement interface.
+// OriginText returns original statement as a string.
 func (a *ExecStmt) OriginText() string {
 	return a.Text
 }
 
-// IsPrepared implements ast.Statement interface.
+// IsPrepared return true if stmt is a prepare statement.
```
Comment (tiancaiamao) [inline executor/adapter.go]: returns

File: executor/adapter.go [hunk 17]
```diff
@@ -155,17 +164,20 @@ type ExecStmt struct {
...
 func (a *ExecStmt) IsPrepared() bool {
 	return a.isPreparedStmt
 }
 
-// IsReadOnly implements ast.Statement interface.
+// IsReadOnly return true if a statement is read only.
```
Comment (tiancaiamao) [inline executor/adapter.go]: ditto

**Comment by zhexuany** [comment]
@tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**